src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable no-console */
+import classNames from 'classnames';
+import React from 'react';
+import { removeTodo } from '../../api/todos';
+import { Errors } from '../../types/Errors';
+import { Filter } from '../../types/Filter';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[]
+  onSetFilterGlobal: React.Dispatch<React.SetStateAction<string>>
+  selectedFilter: string
+  onSetIsError: React.Dispatch<React.SetStateAction<boolean>>
+  onSetTypeError: React.Dispatch<React.SetStateAction<string>>
+  toLoad:() => Promise<void>
+  onSetisDeletedComplete: React.Dispatch<React.SetStateAction<boolean>>
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  onSetFilterGlobal,
+  selectedFilter,
+  onSetIsError,
+  onSetTypeError,
+  toLoad,
+  onSetisDeletedComplete,
+}) => {
+  const amountActiveTodos = todos.filter(
+    todo => todo.completed === false,
+  ).length;
+  const handleAnchorClick = (
+    e: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+    title: string,
+  ) => {
+    e.preventDefault();
+    onSetFilterGlobal(title);
+  };
+
+  const compTodos = todos.filter(todo => todo.completed);
+
+  console.log(compTodos);
+  const clearCompletedTodos = () => {
+    compTodos.forEach(async (a) => {
+      console.log(a);
+      try {
+        onSetisDeletedComplete(true);
+        await removeTodo(a.id);
+      } catch (inError) {
+        console.log('ERROR DELETE', inError);
+        onSetIsError(false);
+        onSetTypeError(Errors.ErrDEL);
+      }
+
+      onSetisDeletedComplete(false);
+      onSetFilterGlobal(Filter.ALL);
+      toLoad();
+    });
+  };
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
```

Comment:
Maybe it will be better to separate components in different way. For example, you don't need to remove all foter from App.tsx. You could make only separate component for **Filter** (this nav tag). And you will pass less props into this component. Also you don't need to move all header in separarte component. You can move only **form for creating new todo**. And this also will help you to don't pass unncessary props. 

And optional recomendation. You can **filter your todos in App.tsx**. And then pass in TodoList already filtered todos. And in your Filter component pass filter status and method to change it.

Reply:
i thought a lot about, and i'm sure it better to keep App as clear as i can. and i tried to imagine that every component have to do separate programer and they need to agree comunication beetwen. create a base structural plan of the APP and after move on. on tech side for this job is not enough context i know (i didn't use it) and maybe something else to imrove working with states.